src/realtime/handlers.ts

```diff
@@ -113,8 +122,13 @@ export class RealtimeSocketHandler {
   private onSocketMessage = (data: RealtimeMessage): void => {
     switch (data.message) {
       case MessagesEnum.RecognitionStarted:
+        if (this.deferDisconnect) {
```

Comment:
In practice I realise `RecognitionStarted` will always be sent first. However since we want to handle the deferred disconnect as soon as the WS connection is established, I wonder if we should run this code outside of the `switch` statement for message types.

Reply:
Do you mean to make the reason for the call more clear? Or do mean to respond to a different event like ws 'connect'?

Reply:
Hooking it into some kind of more generic `connect` event could be good yeah. Otherwise could just have it run right before this `switch` so it runs on the first message (regardless of which type)